test/test_decomp.py

```diff
@@ -415,7 +416,9 @@ def _torch_dispatch(cls, func, types, args=(), kwargs=None):
                 do_relative_check = test_dtype in [torch.float16, torch.bfloat16]
                 real_out_unflat = func(*args, **kwargs)
                 real_out, _ = tree_flatten(real_out_unflat)
-                decomp_out, _ = tree_flatten(decomposition(*args, **kwargs))
+
+                with TorchRefsMode():
```

Comment:
probably we should test decomps as written and with RefsMode? cc @ezyang, @chillee

Reply:
Yes, decomps should work without refs mode. It's important to test this case because refs are specced to be allowed to have more features.

Reply:
Agreed.

Reply:
Just to make sure I understand, are you saying that tests should run decomps both ways (with and without `TorchRefsMode`) and compare both to `real_out`? Is it maybe asking for too much that it works both ways?

If we do want to do this, it effectively turns each test_decomp test into two tests. We'll want to set it up so that we can independently add skips/xfails/etc for each version, right?

Reply:
I definitely think decomps should be tested without refs mode being on. I'm open to decomps being tested with refs mode being on, but this rolls up into a general "does refs mode work" and I think there are other ways to potentially skin this cat. I agree you want separate skips/xfails if you do both.

You should check out the conversation at https://github.com/pytorch/pytorch/pull/82657#discussion_r937022515 ; it contextualizes the importance of TorchRefsMode (the answer being, short term not as important).

Reply:
Thanks for the clarification. I am still unclear why it is important to test the decomps without refs mode (even after reading the thread you linked) but I am still pretty new to the primtorch effort so perfectly happy to just accept it is.

Let me just be explicit about my motivation for this change though: I was adding a ref for `torch.glu`, which called `torch.sigmoid`. When trying to fix some failing `test_decomps.py` tests it took me a while to understand why changes I made to `torch._refs.sigmoid` were not having an effect: namely, the call to `torch.sigmoid` from `torch._refs.glu` was not being remapped to `torch._refs.sigmoid` from inside the tests. This looked wrong to me and I assumed it was not intentional, but I guess you are saying it is?

Reply:
The issue is that we need to register `sigmoid` as a decomposition: https://github.com/pytorch/pytorch/blob/master/torch/_refs/__init__.py#L722

Reply:
Yes this is intentional, because we want to be able to stop decomposing at any point, if some backend can handle torch.sigmoid as written, there's no reason to force it to go to torch._refs.sigmoid. 
Let's roll back a little, what were the changes that you needed to make to torch.sigmoid?

Reply:
Thanks. Since you confirm it is intentional then it seems pretty clear we don't want to make the change in this PR, I'll drop it from the stack.

Re torch.sigmoid, the changes had to do with the precision issues in torch.glu, but let me take another look at it before going into that.